File: js_sdk/src/model/PixelBatchBody.js
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */
import {ApiClient} from '../ApiClient';
import {OpenApiv13pixelbatchBatch} from './OpenApiv13pixelbatchBatch';

/**
 * The PixelBatchBody model module.
 * @module model/PixelBatchBody
 * @version 0.1.1
 */
export class PixelBatchBody {
  /**
   * Constructs a new <code>PixelBatchBody</code>.
   * @alias module:model/PixelBatchBody
   * @class
   * @param batch {Array.<module:model/OpenApiv13pixelbatchBatch>} A list of web events
   * @param pixel_code {String} Pixel ID that can be found in events manager of the TikTok ads platform. For details, see the [Where to Find pixel_code](https://ads.tiktok.com/marketing_api/docs?id=1739584855420929) section
   */
  constructor(batch, pixel_code) {
    this.batch = batch;
    this.pixel_code = pixel_code;
  }

  /**
   * Constructs a <code>PixelBatchBody</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PixelBatchBody} obj Optional instance to populate.
   * @return {module:model/PixelBatchBody} The populated <code>PixelBatchBody</code> instance.
   */
  static constructFromObject(data, obj) {
    if (data) {
      obj = obj || new PixelBatchBody();
      if (data.hasOwnProperty('batch'))
        obj.batch = ApiClient.convertToType(data['batch'], [OpenApiv13pixelbatchBatch]);
      if (data.hasOwnProperty('pixel_code'))
        obj.pixel_code = ApiClient.convertToType(data['pixel_code'], 'String');
    }
    return obj;
  }
}

/**
 * A list of web events
 * @member {Array.<module:model/OpenApiv13pixelbatchBatch>} batch
 */
PixelBatchBody.prototype.batch = undefined;

/**
 * Pixel ID that can be found in events manager of the TikTok ads platform. For details, see the [Where to Find pixel_code](https://ads.tiktok.com/marketing_api/docs?id=1739584855420929) section
 * @member {String} pixel_code
 */
PixelBatchBody.prototype.pixel_code = undefined;

File: java_sdk/src/main/java/business_api_client/PixelBatchBody.java
/*
 * Copyright 2023 TikTok Pte. Ltd.
 *
 * This source code is licensed under the MIT license found in
 * the LICENSE file in the root directory of this source tree.
 */

package business_api_client;

import java.util.Objects;
import java.util.Arrays;
import business_api_client.PixelbatchBatch;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.ArrayList;
import java.util.List;
/**
 * PixelBatchBody
 */

@javax.annotation.Generated(value = "com.tiktok.codegen.JavatiktokcodegenGenerator", date = "2025-07-03T14:36:18.961395-07:00[America/Los_Angeles]")
public class PixelBatchBody {
  @JsonProperty("batch")
  private List<PixelbatchBatch> batch = new ArrayList<PixelbatchBatch>();

  @JsonProperty("pixel_code")
  private String pixelCode = null;

  public PixelBatchBody batch(List<PixelbatchBatch> batch) {
    this.batch = batch;
    return this;
  }

  public PixelBatchBody addBatchItem(PixelbatchBatch batchItem) {
    this.batch.add(batchItem);
    return this;
  }

   /**
   * A list of web events
   * @return batch
  **/
  @Schema(required = true, description = "A list of web events")
  public List<PixelbatchBatch> getBatch() {
    return batch;
  }

  public void setBatch(List<PixelbatchBatch> batch) {
    this.batch = batch;
  }

  public PixelBatchBody pixelCode(String pixelCode) {
    this.pixelCode = pixelCode;
    return this;
  }

   /**
   * Pixel ID that can be found in events manager of the TikTok ads platform. For details, see the [Where to Find pixel_code](https://ads.tiktok.com/marketing_api/docs?id&#x3D;1739584855420929) section
   * @return pixelCode
  **/
  @Schema(required = true, description = "Pixel ID that can be found in events manager of the TikTok ads platform. For details, see the [Where to Find pixel_code](https://ads.tiktok.com/marketing_api/docs?id=1739584855420929) section")
  public String getPixelCode() {
    return pixelCode;
  }

  public void setPixelCode(String pixelCode) {
    this.pixelCode = pixelCode;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    PixelBatchBody pixelBatchBody = (PixelBatchBody) o;
    return Objects.equals(this.batch, pixelBatchBody.batch) &&
        Objects.equals(this.pixelCode, pixelBatchBody.pixelCode);
  }

  @Override
  public int hashCode() {
    return Objects.hash(batch, pixelCode);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class PixelBatchBody {\n");
    
    sb.append("    batch: ").append(toIndentedString(batch)).append("\n");
    sb.append("    pixelCode: ").append(toIndentedString(pixelCode)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}
